lib/Doctrine/DBAL/Query/Expression/ExpressionBuilder.php

```diff
@@ -58,7 +57,7 @@ public function and($expression, ...$expressions) : CompositeExpression
      */
     public function or($expression, ...$expressions) : CompositeExpression
     {
-        return new CompositeExpression(CompositeExpression::TYPE_OR, array_merge([$expression], $expressions));
+        return CompositeExpression::or($expression, ...$expressions);
```

Comment:
These methods do nothing more than forwarding to `CompositeExpression` factory methods. Should we deprecate them and remove them in `4.0`?

Reply:
> Should we deprecate them and remove them in `4.0`?

I don’t think so. `CompositeExpression` is a relatively low-level implementation detail so in most cases the consumers shouldn’t interact with the class directly.